### DISCUSSION TITLE
Querying model for solution availability

### DISCUSSION THREAD

**Discussion by ryanjoneil** [metab0t/PyOptInterface#33] [Q&A] [answered]
If I set a time limit on the solver, once it terminates I would like to know whether or not a feasible solution is available. If a solution is available, then calls to `m.get_value` will return values, otherwise they will raise exceptions.

What is the best way to determine this? Currently I am checking the following condition:

```python
status = m.get_model_attribute(poi.ModelAttribute.PrimalStatus)
if status == poi.ResultStatusCode.FEASIBLE_POINT:
    # we have a solution
```

**Comment by metab0t** [comment, marked as answer]
You are right, checking the status is the correct way.
